Issue 106: analysis pipeline

kaitejohnson opened the pull request:
This PR addresses #106 . It does the following:
- renames the original `_targets.R` file to `_targets_model_run.R`
- creates a new `_targets.R` file which will just rely on the locally saved outputs in `output/` to generate figures and analysis the forecast performance across the locations and forecast dates for which the model has already been run 
- outlines the analysis targets file `_targets.R`
- initially moved the wastewater metadata table creation to this targets file, since there may be additional characteristics/variables we want to extract from the data for the forecast performance analysis (but after realizing this is slow, moved it back and have it loading in in the `_targets.R` pipeline. @ciaramccarthy1 Let me know what you think of this, alternatively, we just make it its own separate pipeline so that we can edit if to add things if needed but don't need to rerun it if we ever need to wipe the analysis targets clean

Still to do (could be in a separate PR)
- create plotting style for the figures 
- write function to create model names for figures
- do some EDA of the scores

## Suggestions for testing:
Once the output data is in the correct folder in `outputs/`, run `tar_make()` and ensure that `tar_load(plot_scores_by_date)` and `tar_load(scatterplot_scores)` generates figures



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

## Release Notes

* **New Features**
  * Added scatterplot visualisation for comparing wastewater plus hospital data against hospital-only data across forecast dates and locations.
  * Introduced restructured analysis pipeline with new configuration-driven workflow stages for improved organisation.

* **Documentation**
  * Updated parameter naming in score conversion documentation for clarity.

* **Refactor**
  * Reorganised pipeline structure to streamline analysis phases and enhance modularity.
  * Updated score processing and aggregation logic.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
eda on scores

Commit:
set up targets pipeline for analysis

Commit:
add function to convert scores to scoringutils object

Commit:
fix docs

Commit:
fix dup

Commit:
fix dup

Commit:
add scatterplot of scores

Commit:
fix linting

Commit:
update ww metadata and move to prev targets script

ciaramccarthy1 commented on R/EDA_plots.R:
```diff
@@ -287,3 +287,36 @@ get_plot_scores_by_date <- function(scores) {
+    ) |>
+    rename(
+      ww_plus_hosp = TRUE,
+      hosp_only = FALSE
```
I added brackets here to avoid this error: `Error in rename():
! Can't rename columns with TRUE.
✖ TRUE must be numeric or character, not TRUE.`

ciaramccarthy1 reviewed (commented):
This worked for me after adding the brackets around TRUE and FALSE in the `get_scatterplot_scores` function. I noticed that the forecast dates in `create_permutations_targets` don't go up to 2025-07-07 (as it says they should do in the comment on lines 26-27) - not sure if this is intentional? Also, what are you thinking re coderabbit's suggestions?

kaitejohnson commented:
> This worked for me after adding the brackets around TRUE and FALSE in the `get_scatterplot_scores` function. I noticed that the forecast dates in `create_permutations_targets` don't go up to 2025-07-07 (as it says they should do in the comment on lines 26-27) - not sure if this is intentional? Also, what are you thinking re coderabbit's suggestions?

Ooh ok I can't remember why but I think this should only go to June 30th, 2025 -- thats all that I have outputs for! 

Have gone through the coderabbit suggestions, fixed the rename, and fixed some bugs in the file.paths!

kaitejohnson commented on _targets_model_run.R:
```diff
@@ -0,0 +1,100 @@
+walk(functions, source)
+rm("functions")
+
+n_workers <- as.integer(floor(future::availableCores() / 4))
+plan(multisession, workers = n_workers)
```
ignoring these

kaitejohnson commented on R/EDA_plots.R:
```diff
@@ -287,3 +287,36 @@ get_plot_scores_by_date <- function(scores) {
     ggtitle("Scores across all locations by forecast dates")
   return(p)
 }
+
+#' Get scatterplot of scores by forecast date and location
+#'
+#' @param scores Data.frame of scores from across locations and forecast dates
+#'
+#' @importFrom ggplot2 geom_bar
+#' @importFrom scoringutils summarise_scores
+#' @returns ggplot object
+#' @autoglobal
+get_scatterplot_scores <- function(scores) {
+  scores_by_forecast <- scores |>
+    summarise_scores(by = c(
+      "model", "include_ww",
+      "hosp_data_real_time", "forecast_date",
+      "location"
+    )) |>
+    filter(model == "wwinference") |>
+    pivot_wider(
+      names_from = include_ww,
+      values_from = wis,
+      id_cols = c(forecast_date, location)
+    ) |>
+    rename(
+      ww_plus_hosp = TRUE,
+      hosp_only = FALSE
+    )
+
+
+  p <- ggplot(scores_by_forecast) +
+    geom_point(aes(x = hosp_only, y = ww_plus_hosp)) +
+    geom_line(aes(x = hosp_only, y = hosp_only), linetype = "dashed")
+  return(p)
+}
```
fixed

kaitejohnson commented on R/prep_scores_to_model.R:
```diff
@@ -0,0 +1,9 @@
+prep_scores_to_model <- function(scores_long,
+                                 ww_metadata) {
+  # Pivot scores from long to wide
+
+  # Join wastewater metadata
+
+  # Placeholder
+  return(NULL)
+}
```
I am going to ignore this for a subsequent PR

kaitejohnson commented on scratch/explore_results.R:
```diff
@@ -0,0 +1,33 @@
+
+scores_overall <- scores |>
+  group_by(model, include_ww) |>
+  summarise(wis = mean(wis)) |>
```
alos ignoring stuff in scratch

kaitejohnson commented on targets/run_gam_targets.R:
```diff
@@ -0,0 +1,17 @@
+    name = scores,
+    command = convert_to_su_object(scores_raw)
+  ),
+  tar_target(
+    name = scores_to_model,
+    command = prep_scores_to_model(
+      scores_long = scores,
+      ww_metadata = ww_metadata_table_combines
+    )
+  )
```
This is a placeholder function to be filled in!

Commit:
fix filepaths and rename bug

ciaramccarthy1 commented:
Re the metadata table @kaitejohnson, the analysis pipeline is running much more quickly now so I think its okay to leave it as is (versus having that in a separate pipeline).
